slatedb/src/compactor.rs

```diff
@@ -1047,11 +1057,25 @@ mod tests {
         (manifest_store, table_store)
     }
 
-    async fn await_compaction(manifest_store: Arc<ManifestStore>) -> Option<CoreDbState> {
+    /// Waits until all writes have made their way to L1 or below. No data is allowed in
+    /// in-memory WALs, in-memory memtables, or L0 SSTs on object storage.
+    async fn await_compaction(db: &Db, manifest_store: Arc<ManifestStore>) -> Option<CoreDbState> {
         run_for(Duration::from_secs(10), || async {
-            let db_state = get_db_state(manifest_store.clone()).await;
-            if db_state.l0_last_compacted.is_some() {
-                return Some(db_state);
+            let (empty_wal, empty_memtable) = {
+                let mut db_state = db.inner.state.write();
```

Comment:
nit: can use a read lock here

Reply:
Unfortunately, both `wal()` and `memtable()` require mut:

```
    pub fn wal(&mut self) -> &mut WritableKVTable {
        &mut self.wal
    }

    pub fn memtable(&mut self) -> &mut WritableKVTable {
        &mut self.memtable
    }
```

If I change `write()` to `read()`, I get:

```
cannot borrow data in dereference of `parking_lot::lock_api::RwLockReadGuard<'_, parking_lot::RawRwLock, DbState>` as mutable
trait `DerefMut` is required to modify through a dereference, but it is not implemented for `parking_lot::lock_api::RwLockReadGuard<'_, parking_lot::RawRwLock, DbState>`
```